econml/metalearners.py

```diff
@@ -83,10 +89,16 @@ def effect(self, X):
         """
         # Check inputs
         X = check_array(X)
-        tau_hat = self.treated_model.predict(X) - self.controls_model.predict(X)
+        # Check treatment
+        if T0 not in self.unique_T or T1 not in self.unique_T:
+            raise ValueError(
+                "T0 and T1 must be scalers in T you passed in at fitting time, please pass in values from {}".format(self.unique_T))
+        ind1, = np.where(self.unique_T == T1)[0]
+        ind0, = np.where(self.unique_T == T0)[0]
+        tau_hat = self.models[ind1].predict(X) - self.models[ind0].predict(X)
         return tau_hat
 
-    def marginal_effect(self, X):
+    def marginal_effect(self, X, T1):
```

Comment:
When a method takes only one treatment, probably better to just call it `T`.

```suggestion
    def marginal_effect(self, X, T):
```